packages/frontend/src/components/chat/ChatContextMenu.tsx

```diff
@@ -0,0 +1,516 @@
+        label: tx('menu_view_profile'),
+        action: onViewProfile,
+      },
+    selfInGroup && {
+      label: tx('menu_view_profile'),
+      dataTestid: 'edit-group',
+      action: onViewGroup,
+    },
+    // Edit Channel
+    singleChat.chatType === 'OutBroadcast' && {
+      label: tx('edit_channel'),
+      action: onViewGroup,
+    },
+  ]
+}
+
+/**
+ * Builds encryption info menu item
+ */
+function buildEncryptionInfoMenuItem(
+  singleChat: ChatListItem | false,
+  tx: ReturnType<typeof useTranslationFunction>,
+  openEncryptionInfoDialog: (info: {
+    chatId: number | null
+    dmChatContact: number | null
+  }) => void
+): ContextMenuItem | false {
+  if (!singleChat) return false
+
+  return (
+    !singleChat.isDeviceTalk &&
+    !singleChat.isSelfTalk && {
+      label: tx('encryption_info_title_desktop'),
+      action: () =>
+        openEncryptionInfoDialog({
+          chatId: singleChat.id,
+          dmChatContact: singleChat.dmChatContact,
+        }),
+    }
+  )
+}
+
+/**
+ * provides a context menu for chat list items
+ * and for the 3dot menu in main chat view
+ */
+export function useChatContextMenu(): {
+  /**
+   * @param chatListItems array of selected chat list items (empty if opened from main chat view)
+   * @param selectedChatId id of the currently selected chat (null if none selected)
+   * @param selectedChat the currently active chat (only in main view)
+   */
+  openContextMenu: (
+    event: React.MouseEvent<any, MouseEvent>,
+    chatListItems: ChatListItem[],
+    selectedChatId: number | null,
+    selectedChat?: T.FullChat
+  ) => Promise<void>
+  activeContextMenuChatIds: number[]
+} {
+  const { openDialog } = useDialog()
+  const {
+    openBlockFirstContactOfChatDialog,
+    openEncryptionInfoDialog,
+    openDeleteChatsDialog,
+    openLeaveGroupOrChannelDialog,
+    openClearChatDialog,
+  } = useChatDialog()
+  const openViewGroupDialog = useOpenViewGroupDialog()
+  const openViewProfileDialog = useOpenViewProfileDialog()
+  const { openContextMenu } = useContext(ContextMenuContext)
+  const accountId = selectedAccountId()
+  const { unselectChat } = useChat()
+  const tx = useTranslationFunction()
+  const [activeContextMenuChatIds, setActiveContextMenuChatIds] = useState<
+    number[]
+  >([])
+
+  const openContextMenuHandler = async (
+    event: React.MouseEvent<any, MouseEvent>,
+    chatListItems: ChatListItem[],
+    selectedChatId: number | null,
+    selectedChat?: T.FullChat
+  ) => {
+    if (chatListItems.length === 0 && selectedChat === undefined) {
+      log.error('openContextMenu called with 0 chats')
+      return
+    }
+    const isMainView = !!selectedChat
+    const singleChat: ChatListItem | false = isMainView
+      ? fullChatToChatListItem(selectedChat)
+      : chatListItems.length === 1
+        ? chatListItems[0]
+        : false
+
+    if (chatListItems.length === 0 && singleChat) {
+      chatListItems.push(singleChat)
+    }
```

Comment:
I think mutating arguments will cause a bug at some point, unless there is a bug already.